src/website/app/Markdown.js

```diff
@@ -0,0 +1,182 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+import React, { createElement } from 'react';
+import dedent from 'dedent';
+import marksy from 'marksy/components';
+import { createStyledComponent } from '../../utils';
+import Heading from './Heading';
+import Link from './Link';
+import prism from './utils/prism';
+
+type Props = {
+  children: string,
+  className?: string,
+  scope?: {
+    [string]: MnrlReactNode
+  }
+};
+
+type mdCodeProps = {
+  language?: string,
+  children?: string,
+  code?: string
+};
+
+type mdHeadingProps = {
+  id?: string
+};
+
+type mdLinkProps = {
+  href: string,
+  title?: string,
+  target?: string,
+  children: MnrlReactNode
+};
+
+const Root = createStyledComponent('div', ({ theme }) => ({
+  lineHeight: theme.lineHeight_prose,
+
+  '& a:link': {
+    color: theme.color_text_primary,
+    textDecoration: 'none'
+  },
+  '& a:hover': {
+    color: theme.color_text_primary_hover,
+    textDecoration: 'underline'
+  },
+  '& a:focus': {
+    color: theme.color_text_primary_focus,
+    outline: `1px solid ${theme.borderColor_focus}`,
+    outlineOffset: '2px'
+  },
+  // `:active` must be last, to follow LVHFA order:
+  // https://developer.mozilla.org/en-US/docs/Web/CSS/:active
+  '& a:active': {
+    color: theme.color_text_primary_active
+  },
+
+  '& :not(pre) > code': {
+    backgroundColor: theme.color_gray_20,
+    borderRadius: theme.borderRadius_1,
+    fontFamily: theme.fontFamily_monospace,
+    padding: theme.spacing_half
+  },
+
+  // Specificity silliness due to having to style markdown content off of the
+  // container
+  '& h2,& h3,& h4,& h5,& h6': {
+    '& > a:link': {
+      color: theme.color_caption
+    },
+    '& > a:hover': {
+      color: theme.color_text_primary_hover
+    },
+    '& > a:focus': {
+      color: theme.color_text_primary_focus
+    },
+    // `:active` must be last, to follow LVHFA order:
+    // https://developer.mozilla.org/en-US/docs/Web/CSS/:active
+    '& > a:active': {
+      color: theme.color_text_primary_active
+    }
+  },
+
+  '& pre': {
+    fontSize: theme.fontSize_ui,
+    maxHeight: `${parseFloat(theme.spacing_quad) * 10}em`,
+    overflow: 'auto'
+  }
+}));
+
+function replaceHeading(level, children, headingProps: mdHeadingProps) {
+  // Render the same props and children that were passed in, but prepend a
+  // link to this title with the text '#'.
+  return (
+    <Heading level={level} id={headingProps.id}>
+      {children}
+    </Heading>
+  );
+}
+
+function isRoutedLink(url) {
+  // If url doesn't start with `#` or contain `.`, then we use a routed Link
+  return !RegExp(/(^#|\.)/).test(url);
+}
+
+export default function Markdown({ children, className, scope }: Props) {
+  const compile = marksy({
+    createElement,
+    elements: {
+      a({ href, children }: mdLinkProps) {
+        return isRoutedLink(href)
```

Comment:
This could maybe be simplified. e.g.

```js
const linkProps = isRoutedLink(href) ? { to: href } : { href };
return <Link {...linkProps}>{children}</Link>
```